staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -244,14 +365,33 @@ func IsGroupDiscoveryFailedError(err error) bool {
 }
 
 func ServerGroupsAndResources(d DiscoveryInterface) ([]*metav1.APIGroup, []*metav1.APIResourceList, error) {
-	sgs, err := d.ServerGroups()
+	var sgs *metav1.APIGroupList
+	var resources []*metav1.APIResourceList
+	var err error
+
+	// If the passed discovery object implements the wider AggregatedDiscoveryInterface,
+	// then it is attempt to retrieve both the groups and the resources.
+	ad, ok := d.(AggregatedDiscoveryInterface)
+	if ok {
```

Comment:
given that `ad` is not used outside the Xcode of the if, it would be cleaner to combine the condition and declaration

```suggestion
	if ad, ok := d.(AggregatedDiscoveryInterface); ok {
```

Reply:
Fixed here and in `CachedDiscoveryClient`